test/distributed/org/apache/cassandra/distributed/mock/nodetool/NetstatsTest.java

```diff
@@ -0,0 +1,607 @@
+                    }
+                    else if (alreadySent >= o.alreadySent
+                             && progressFiles >= o.progressFiles
+                             && bytesTotalSoFar > o.bytesTotalSoFar
+                             && progressBytes >= o.progressBytes)
+                    {
+                        return 1;
+                    }
+                    else
+                    {
+                        throw new IllegalStateException(String.format("Could not compare arguments %s and %s", this, o));
+                    }
+                }
+            }
+
+            public static class SendingSSTable
+            {
+                private static final Pattern sendingFilePattern = Pattern.compile("(.*) (.*)/(.*) bytes \\((.*)%\\) sent to (.*)");
+
+                long bytesSent = 0;
+                long bytesInTotal = 0;
+                double progress = 0.0f;
+
+                public static SendingSSTable parseTable(String table)
+                {
+                    final Matcher matcher = sendingFilePattern.matcher(table);
+
+                    if (matcher.matches())
+                    {
+                        final SendingSSTable sendingSSTable = new SendingSSTable();
+
+                        sendingSSTable.bytesSent = Long.parseLong(matcher.group(2));
+                        sendingSSTable.bytesInTotal = Long.parseLong(matcher.group(3));
+                        sendingSSTable.progress = Double.parseDouble(matcher.group(4));
+
+                        return sendingSSTable;
+                    }
+
+                    throw new IllegalStateException("Table does not match - " + table);
+                }
+
+                public String toString()
+                {
+                    return "SendingSSTable{" +
+                           "bytesSent=" + bytesSent +
+                           ", bytesInTotal=" + bytesInTotal +
+                           ", progress=" + progress +
+                           '}';
+                }
+            }
+        }
+    }
+
+    private static final class NetstatResults
+    {
+        private final List<String[]> netstatOutputs = new ArrayList<>();
+
+        public void add(String[] result)
+        {
+            netstatOutputs.add(result);
+        }
+
+        public void assertSuccessful()
+        {
+            for (final String[] result : netstatOutputs)
+            {
+                InstanceAwareNodetool.assertSuccessful(result);
+            }
+        }
+    }
+
+    private static class NetstatsCallable implements Callable<NetstatResults>
+    {
+        private final IInvokableInstance node;
+        private final InstanceAwareNodetool nodetool;
+
+        public NetstatsCallable(final InstanceAwareNodetool nodetool,
+                                final IInvokableInstance node)
+        {
+            this.nodetool = nodetool;
+            this.node = node;
+        }
+
+        public NetstatResults call() throws Exception
```

Comment:
FYI there is a class `org.apache.cassandra.utils.Retry` which can do retries for you, the main benefit is it will gradually backoff rather than retry every 1s.

In this case I am ok with this, just giving as an FYI